lib/ddtrace/contrib/httprb/patcher.rb

```diff
@@ -0,0 +1,37 @@
+require 'ddtrace/contrib/patcher'
+require 'ddtrace/contrib/httprb/instrumentation'
+require 'ddtrace/contrib/httprb/datadog_wrap'
+
+module Datadog
+  module Contrib
+    # Datadog Httprb integration.
+    module Httprb
+      # Patcher enables patching of 'httprb' module.
+      module Patcher
+        include Contrib::Patcher
+
+        module_function
+
+        def patched?
+          done?(:httprb)
+        end
+
+        # patch applies our patch
+        def patch
+          do_once(:httprb) do
+            begin
+              register_feature!
+              ::HTTP::Options.send(:include, Instrumentation)
+            rescue StandardError => e
+              Datadog::Tracer.log.error("Unable to apply httprb integration: #{e}")
+            end
+          end
+        end
+
+        def register_feature!
+          ::HTTP::Options.register_feature(:datadog_wrap, Datadog::Contrib::Httprb::DatadogWrap)
```

Comment:
@ericmustin Looks like some of the latest tests for 2.0 and 2.1 don't work with this: `(/app/lib/ddtrace/contrib/httprb/patcher.rb:26:in `rescue in block in patch') Unable to apply httprb integration: undefined method `register_feature' for HTTP::Options:Class`

Is an older version of HTTP.rb running for these Ruby versions? Which don't have these? Seems likely given they're older.

Reply:
yup that's corrrect, i've moved away from this `register_feature` approach now tho (in favor of monkeypatching) , so able to support back to 2.0. i think this is good to resolve but feel free to reopen if i'm mistaken